themes/openy_themes/openy_rose/openy_rose.theme

```diff
@@ -119,38 +136,170 @@ function openy_rose_form_system_theme_settings_alter(array &$form, FormStateInte
+    '#description' => t('To display a logo in the footer area, upload it here.</br>Image size suggested 99 x 75 pixels.'),
     '#upload_validators' => array(
-      'file_validate_extensions' => array("png jpg jpeg gif svg"),
+      'file_validate_extensions' => array('png jpg jpeg gif svg'),
     ),
     '#upload_location' => 'public://',
     '#default_value' => theme_get_setting('openy_rose_footer_logo'),
   );
+  $form['camp_section'] = array(
+    '#type' => 'details',
+    '#title' => t('Camp section'),
+    '#open' => TRUE,
+  );
+  $form['camp_section']['openy_rose_camp_section_logo'] = array(
+    '#type' => 'managed_file',
+    '#title' => t('Camp section logo'),
+    '#description' => t('To display a logo in the header area on camp pages, and camp related landing pages, upload it here.</br>Image size suggested 92 x 70 pixels.'),
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('png jpg jpeg gif svg'),
+    ),
+    '#upload_location' => 'public://',
+    '#default_value' => theme_get_setting('openy_rose_camp_section_logo'),
+  );
+  $form['camp_section']['openy_rose_camp_footer_logo'] = array(
+    '#type' => 'managed_file',
+    '#title' => t('Camp section footer logo'),
+    '#description' => t('To display a logo in the footer area on camp pages, and camp related landing pages, upload it here.</br>Image size suggested 99 x 75 pixels.'),
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('png jpg jpeg gif svg'),
+    ),
+    '#upload_location' => 'public://',
+    '#default_value' => theme_get_setting('openy_rose_camp_footer_logo'),
+  );
+  $form['camp_section']['openy_rose_camp_favicon'] = array(
+    '#type' => 'managed_file',
+    '#title' => t('Camp favicon'),
+    '#description' => t('To display a favicon on camp pages, and camp related landing pages, upload it here.</br>Image sizes must be 16×16, 32×32, 48×48, or 64×64 pixels.</br>Accepted file types ico, png, gif.'),
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('ico png gif'),
+    ),
+    '#upload_location' => 'public://',
+    '#default_value' => theme_get_setting('openy_rose_camp_favicon'),
+  );
 
   // Submission handler is not working on theme form
   // See https://www.drupal.org/node/1862892 & https://www.drupal.org/node/2779947
-  // Setting the status to saved permanently.
-  $image_custom_index = theme_get_setting('openy_rose_footer_logo');
-  if (isset($image_custom_index)) {
-    $fid = theme_get_setting('openy_rose_footer_logo');
-    if (!empty($fid) && is_array($fid) && is_numeric(reset($fid))) {
-      $file = File::load(reset($fid));
-      if (!$file->isPermanent()) {
-        // Set logo permanent flag and save.
-        $file->setPermanent();
-        $file->save();
+  // Setting file usage for openy_rose theme.
+  $image_fields = [
+    'openy_rose_footer_logo',
```

Comment:
This looks like a config. Is it possible to save/get this from config somehow and get rid of hardcoded list of values?
Also it is possible to get list of image fields dynamically. BTW, this looks like a follow-up tasks, not the do it now.